src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -319,7 +319,13 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                 container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Work), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Work)),
                     readOnly: container.isReadOnlyVolume(Constants.DefaultContainerMounts.Work)));
             }
-
+            
+            if (AgentKnobs.AllowMountTasksReadonlyOnWindows.GetValue(executionContext).AsBoolean())
```

Comment:
If it's true - it will be mounted second time for non windows platform - this contradicts knob description.
Could it make sense to make this behavior windows only?